client/dashboard/sites/overview-domains-card/index.tsx

```diff
@@ -1,13 +1,107 @@
-import { DomainSubtype } from '@automattic/api-core';
-import { domainsQuery } from '@automattic/api-queries';
+import { DomainSubtype, type DomainSummary, type Site } from '@automattic/api-core';
+import { domainsQuery, siteCurrentPlanQuery } from '@automattic/api-queries';
 import { useQuery } from '@tanstack/react-query';
+import { useRouter } from '@tanstack/react-router';
+import { DataViews, filterSortAndPaginate, View } from '@wordpress/dataviews';
+import { __ } from '@wordpress/i18n';
+import { useState, useMemo } from 'react';
+import { siteDomainsRoute } from '../../app/router/sites';
+import { CalloutSkeleton } from '../../components/callout-skeleton';
+import { Card, CardHeader, CardBody } from '../../components/card';
+import RouterLinkButton from '../../components/router-link-button';
+import { SectionHeader } from '../../components/section-header';
+import { useFields, DEFAULT_VIEW, DEFAULT_LAYOUTS } from '../../domains/dataviews';
 import { isTransferrableToWpcom } from '../../utils/domain-types';
 import { isSelfHostedJetpackConnected } from '../../utils/site-types';
 import DomainTransferUpsellCard from '../overview-domain-transfer-upsell-card';
 import DomainUpsellCard from '../overview-domain-upsell-card';
-import type { Site } from '@automattic/api-core';
+import type { DomainsView } from '../../domains/dataviews';
 
-export default function DomainsCard( { site }: { site: Site } ) {
+const getDomainId = ( domain: DomainSummary ): string => {
+	return `${ domain.domain }-${ domain.blog_id }`;
+};
+
+const SiteDomainDataViews = ( {
+	site,
+	domains,
+	type,
+}: {
+	site: Site;
+	domains: DomainSummary[];
+	type: DomainsView[ 'type' ];
+} ) => {
+	const router = useRouter();
+	const fields = useFields( { site, inOverview: true } );
+
+	const [ initialView, setView ] = useState< DomainsView >( {
+		...DEFAULT_VIEW,
+		type,
+	} );
+
+	const view = useMemo(
+		() => ( {
+			...initialView,
+			type,
+			fields: [ 'ssl_status', 'domain_status' ],
+		} ),
+		[ initialView, type ]
+	);
+
+	const { data: filteredData, paginationInfo } = filterSortAndPaginate(
+		domains,
+		view as View,
+		fields
+	);
+
+	return (
+		<Card>
+			<CardHeader
+				style={ {
+					flexDirection: 'column',
+					alignItems: 'stretch',
+				} }
+			>
+				<SectionHeader
+					title={ __( 'Domains' ) }
+					level={ 3 }
+					actions={
+						<RouterLinkButton
+							variant="secondary"
+							size="compact"
+							to={
+								router.buildLocation( {
```

Comment:
This is not working in v1.

Reply:
Noting that when I click a `wpcomstaging` domain, I get a subscription error afterwards. 

Should it be clickable?

![Image](https://github.com/user-attachments/assets/d0c42e21-3e48-48a5-8baf-30421dcb9353)

Reply:
Decided not to backport the card into v1. See https://github.com/Automattic/wp-calypso/pull/106849#discussion_r2485893803 for reasoning.

> Noting that when I click a wpcomstaging domain, I get a subscription error afterwards.
Should it be clickable?

According to p1762162866602729/1762134664.961369-slack-C0982082MSR, no.